backend/src/main/java/gov/cdc/usds/simplereport/config/ApiUserAwareGraphQlContextBuilder.java

```diff
@@ -111,7 +119,22 @@ private DataLoaderRegistry buildDataLoaderRegistry() {
                           .map(p -> found.getOrDefault(p, PatientPreferences.DEFAULT))
                           .collect(Collectors.toList());
                     }));
-    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesLoader);
+    dataLoaderRegistry.register(PatientPreferences.DATA_LOADER, patientPreferencesDataLoader);
+
+    DataLoader<UUID, TestEvent> testEventDataLoader =
+        new DataLoader<>(
+            patientIds ->
+                supplyAsync(
+                    () -> {
+                      Map<UUID, TestEvent> found =
+                          testEventRepository.findLastTestsByPatient(patientIds).stream()
```

Comment:
I'm confused why this works. It clearly does... but it's also clearly relying on being called within a transaction and I'm not sure who is managing that transaction.

Reply:
I naively assumed that this was bundled into whatever transaction was being used by the GraphQL context to generally resolve its data. Any lead as to where I might track this down to confirm?

Reply:
You're clearly right as to what's happening, I'm just a little unnerved by the strange action at a distance involved. Not something to necessarily put a lot of attention into right now, but ... it's a little odd.  Normally our transactions are all managed at the service layer, and this seems explicitly to be at the resolver layer, and yet here we are?